java/src/jmri/util/FileUtilSupport.java

```diff
@@ -485,19 +510,25 @@ public boolean isPortableFilename(String filename) {
      *
      * @return User's home directory as a String
      */
+    @Nonnull
+    @CheckReturnValue
     public String getHomePath() {
         return HOME_PATH;
     }
 
     /**
      * Get the user's files directory. If not set by the user, this is the same
-     * as the profile path.
+     * as the profile path; if all else fails, the program directory. 
```

Comment:
This should be the settings directory, not the program directory, since this is a path that JMRI expects to write to.